src/featureSupport.ts

```diff
@@ -32,39 +37,59 @@ export class GitpodVersion {
 }
 
 let cacheGitpodVersion: { host: string; version: GitpodVersion } | undefined;
-async function getOrFetchVersionInfo(serviceUrl: string) {
+async function getOrFetchVersionInfo(serviceUrl: string, logger: Log) {
     if (serviceUrl === 'https://gitpod.io') {
-        return undefined;
+        return {
+            // SaaS default allow all features, should proper handle SaaS feature support if needed in the future
+            version: GitpodVersion.Max,
+        };
     }
 
     if (serviceUrl === cacheGitpodVersion?.host) {
+        logger.info(`Using cached version ${cacheGitpodVersion.version} for ${serviceUrl}`);
         return cacheGitpodVersion;
     }
 
-    let gitpodRawVersion: string;
-    try {
-        const versionEndPoint = `${serviceUrl}/api/version`;
-        const versionResponse = await fetch(versionEndPoint);
-        if (!versionResponse.ok) {
-            return undefined;
+    const fetchVersion = async (times: number = 3): Promise<string | undefined> => {
+        try {
+            const versionEndPoint = `${serviceUrl}/api/version`;
+            const resp = await fetch(versionEndPoint, { timeout: 1500 });
+            if (!resp.ok) {
+                throw new Error(`Response with ${resp.status} ${resp.statusText}`);
+            }
+            return await resp.text();
+        } catch (e) {
+            logger.error(`Failed to fetch version with from: ${serviceUrl} left attempt ${times - 1}`, e);
+            if (times - 1 > 0) {
+                await new Promise(resolve => setTimeout(resolve, 1000));
+                return fetchVersion(times - 1);
+            } else {
+                return undefined;
+            }
         }
+    };
 
-        gitpodRawVersion = await versionResponse.text();
-    } catch (e) {
-        return undefined;
+    const gitpodRawVersion = await fetchVersion();
+    if (!gitpodRawVersion) {
+        return {
```

Comment:
Maybe add one more log here telling we are using min version as default

Reply:
Added